src/Microsoft.Health.Fhir.Shared.Core/Features/Resources/Delete/DeletionService.cs

```diff
@@ -192,7 +193,24 @@ public async Task<long> DeleteMultipleAsync(ConditionalDeleteResourceRequest req
                 await cancellationTokenSource.CancelAsync();
             }
 
-            System.Threading.Tasks.Task.WaitAll(deleteTasks.ToArray(), cancellationToken);
+            try
+            {
+                // We need to wait until all running tasks are cancelled to get a count of resources deleted.
+                Task.WaitAll(deleteTasks.ToArray(), cancellationToken);
+            }
+            catch (AggregateException age)
+            {
+                // If one of the tasks fails, the rest may throw a cancellation exception. Filtering those out as they are noise.
+                foreach (var coreException in age.InnerExceptions.Where(e => e is not TaskCanceledException))
+                {
+                    _logger.LogError(age, "Error deleting");
```

Comment:
This is logging the aggregate exception multiple times. I think you want to log coreException.

Reply:
wow thank you need coffee